ui-tests/tests/widget.test.ts

```diff
@@ -179,6 +179,410 @@ ConnectorWidget()`)
+      }) => {
+            await displayWidget(page);
+
+            await page.locator("#createNewConnection").click();
+            await page
+              .locator("#selectConnection")
+              .selectOption({ label: "PostgreSQL" });
+            await page.locator("#username").fill("someuser");
+            await page.locator("#password").fill("somepassword");
+            await page.locator("#host").fill("localhost");
+            await page.locator("#port").fill("5432");
+            await page.locator("#database").fill("somedb");
+            await page.locator("#selectConnection").selectOption({ label: label });
+
+            expect(
+              await page.locator(`#connectionName`).evaluate((select) => select.value)
+            ).toBe(connectionName);
+            expect(await page.locator(`#port`).evaluate((select) => select.value)).toBe(
+              port
+            );
+            expect(
+              await page.locator(`#username`).evaluate((select) => select.value)
+            ).toBe(username);
+            expect(
+              await page.locator(`#password`).evaluate((select) => select.value)
+            ).toBe(password);
+            expect(await page.locator(`#host`).evaluate((select) => select.value)).toBe(
+              host
+            );
+            expect(
+              await page.locator(`#database`).evaluate((select) => select.value)
+            ).toBe(database);
+      });
+}
+
+const autoPopulatedFieldsExistingConnection = [
+  { label: "MySQL", connectionName: "mysql", port: "3306", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "MariaDB", connectionName: "mariadb", port: "3306", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "Snowflake", connectionName: "snowflake", port: "443", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "Oracle", connectionName: "oracle", port: "1521", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "MSSQL", connectionName: "mssql", port: "1433", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "Redshift", connectionName: "redshift", port: "5439", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+];
+
+for (const { label, connectionName, database, port, username, password, host } of autoPopulatedFieldsExistingConnection) {
+   test(`test fields are auto-populated if dropdown selection changes, existing connection
+       present, default DB alias displayed: ${label}`, async ({
+        page,
+      }) => {
+        await createNewNotebook(page);
+
+        await page.notebook.enterCellEditingMode(0);
+        const cell = await page.notebook.getCell(0);
+        await cell?.type(`
+from pathlib import Path
```

Comment:
I'm unsure about this test. why would having a connections.ini change anything? or are you testing that a connection.ini does not change behavior?

Reply:
If user doesn't change the default connection alias and changes the dropdown there would be two cases:
1. If some connection does not exist, the alias will be `default`.
2. If some other connection exists, the alias will be DB specific, e.g. `oracle` for Oracle DB.

This case is for the 2nd scenario

Reply:
right, i forgot about that scenario!